contracts/NonfungiblePositionManagerV4.sol

```diff
@@ -120,6 +121,15 @@ contract NonfungiblePositionManagerV4 is
         // TODO: implement this
     }
 
+    /// @dev Caches a pool key
+    function cachePoolKey(PoolKey memory poolKey) private returns (PoolId poolId) {
+        poolId = poolKey.toId();
+        // uninitialized check
+        if (_poolIdToPoolKey[PoolId.unwrap(poolId)].tickSpacing == 0) {
```

Comment:
is checking for `tickSpacing` the right thing to do here? just want to check if this key exists in the mapping yet

Reply:
also big difference between this mapping vs the one that exists in v3:
- in v3, there's a poolId indicator PER nft so during burn, we deleted the poolId from the mapping
- but in v4, this is a poolId PER poolkey and there could be multiple NFTs per poolKey. we don't want to burn during mint, so this is essentially an ever-growing list. but it also means not every `mint` needs to store an extra poolId mapping

Reply:
yea i think its a fair chec as `tickSpacing` can't be 0 in an initialized pool